From e5c9f09ff66859d85da07e0704cad617539408d6 Mon Sep 17 00:00:00 2001
From: Andi Gerl <andi.gerl@exacttechnology.com>
Date: Tue, 5 Mar 2024 11:09:47 -0500
Subject: [PATCH] net: lib: lwm2m: Use int16_t for signal quality

Allow negative values for RSRQ, duplicated from 8e4c588eab9828e99ca0bf696982c1275bb13229

RSRQ is the ratio between send and received signal strength and usually
understood/expected to be represented as a ratio in dB and as such always
has a negative range. So to allow RSRQ to be represented correctly the
resource must allow negative values, but currently it's limited to unsigned
8bit integers.
---
 subsys/net/lib/lwm2m/lwm2m_obj_connmon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/subsys/net/lib/lwm2m/lwm2m_obj_connmon.c b/subsys/net/lib/lwm2m/lwm2m_obj_connmon.c
index c0e14429c85..f59940ceae9 100644
--- a/subsys/net/lib/lwm2m/lwm2m_obj_connmon.c
+++ b/subsys/net/lib/lwm2m/lwm2m_obj_connmon.c
@@ -85,7 +85,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 /* resource state variables */
 static int8_t net_bearer;
 static int8_t rss;
-static uint8_t link_quality;
+static int16_t link_quality;
 static uint32_t cellid;
 static uint16_t mnc;
 static uint16_t mcc;
@@ -100,7 +100,7 @@ static struct lwm2m_engine_obj_field fields[] = {
 	OBJ_FIELD_DATA(CONNMON_NETWORK_BEARER_ID, R, U8),
 	OBJ_FIELD_DATA(CONNMON_AVAIL_NETWORK_BEARER_ID, R, U8),
 	OBJ_FIELD_DATA(CONNMON_RADIO_SIGNAL_STRENGTH, R, S8),
-	OBJ_FIELD_DATA(CONNMON_LINK_QUALITY, R, U8),
+	OBJ_FIELD_DATA(CONNMON_LINK_QUALITY, R, S16),
 	OBJ_FIELD_DATA(CONNMON_IP_ADDRESSES, R, STRING),
 	OBJ_FIELD_DATA(CONNMON_ROUTER_IP_ADDRESSES, R_OPT, STRING),
 	OBJ_FIELD_DATA(CONNMON_LINK_UTILIZATION, R_OPT, U8),
-- 
2.34.1

